src/components/HomeSection.test.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import HomeSection from './HomeSection';
+
+describe('HomeSection', () => {
+  test('checks if the title is in the document', () => {
+    render(
+      <HomeSection
+        title="Topology"
+        description="Math course"
+        buttons={[]}
+        pic="manifold"
+      />
+    );
+    screen.debug();
+    expect(screen.getByText('Topology')).toBeInTheDocument();
+  });
+
+  test('checks if the description is in the document', () => {
+    render(
+      <HomeSection
+        title="Topology"
+        description="Math course"
+        buttons={[]}
+        pic="manifold"
+      />
+    );
+    expect(screen.getByText('Math course')).toBeInTheDocument();
+  });
+
+  test('checks if it renders the given number of buttons', () => {
+    render(
+      <HomeSection
+        title="Topology"
+        description="Math course"
+        buttons={[]}
+        pic="manifold"
+      />
+    );
+    screen.debug();
+    const items = screen.queryByRole('button');
+    expect(items).toBe(null);
```

Comment:
Good job identifying to use ***ByRole**!

Reply:
Btw, since `queryByRole` only returns one element I suggest replacing this with:

```js
    const item = screen.queryByRole('button');
    expect(item).toBe(null);
```
or:
```js
    const items = screen.queryAllByRole('button');
    expect(items.length).toBe(0);
```